module/VuFind/src/VuFind/Search/Factory/EPFBackendFactory.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/**
+ * Factory for EDS backends used for EPF.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2013.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Search
+ * @author   David Maus <maus@hab.de>
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+
+namespace VuFind\Search\Factory;
+
+/**
+ * Factory for EDS backends used for EPF.
+ *
+ * @category VuFind
+ * @package  Search
+ * @author   David Maus <maus@hab.de>
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+class EPFBackendFactory extends EdsBackendFactory
+{
+    /**
+     * Get the service name.  This is used for both configuration
+     * and record driver retrieval.
+     *
+     * @return str
+     */
+    protected function getServiceName()
+    {
+        return 'EPF';
+    }
+
+    /**
+     * Create the options array for the EDS connector.
+     *
+     * @return array
+     */
+    protected function createConnectorOptions()
+    {
+        $options = parent::createConnectorOptions();
+        $options['api_url'] = $this->edsConfig->General->api_url
+            ?? 'https://eds-api.ebscohost.com/edsapi/publication';
+
+        if (isset($this->edsConfig->General->session_url)) {
+            $options['session_url'] = $this->edsConfig->General->session_url;
```

Comment:
Any reason not to add this option to the base factory and EDS.ini as well? If we're making it configurable, I don't see harm in having the same configuration work in both .ini files. Combined with the above suggestion, that might eliminate the need for this whole custom method.

Reply:
Yes, done.  Originally I had only added sessionURL on the EPF side; on the EDS side I reused the API URL value for session since it was identical.  That worked but I'd rather just be specific and cleaner about it.  So I also removed some logic I had added in createSession() that was falling back to the apiHost if sessionHost was not present -- not needed since we now have a default.

Reply:
Well, to clarify, I left the createConnectorOptions() method in EdsBackendFactory, but no more need to override. in the EPF version.  I think that's ok because I can see possibly needing to override it in the future, and it breaks out a somewhat complex part of the connector setup.  But can move it back into createConnector() if you prefer.

Reply:
I have no objection with leaving this refactoring in place -- as you say, it has potential benefits even if we don't need to leverage it for EPF.